core/shared/src/main/scala/com/lynbrookrobotics/potassium/tasks/AndUntilDoneTask.scala

```diff
@@ -0,0 +1,31 @@
+package com.lynbrookrobotics.potassium.tasks
+
+/**
+  * A finite task where a continuous task is run while a finite task is running
+ *
+  * @param first the finite task to run
+  * @param second the continuous task to run at the same time
+  */
+class AndUntilDoneTask private[tasks](first: FiniteTask, second: ContinuousTask)
+  extends FiniteTask with FiniteTaskFinishedListener {
+  override def onFinished(task: FiniteTask): Unit = {
+    if (!first.isRunning) {
+      finished()
```

Comment:
Won't this cause endless recursion? Is that the point, it keeps going until first finishes?